src/TestSuite/MappingGenerator.php

```diff
@@ -0,0 +1,131 @@
+<?php
+declare(strict_types=1);
+
+namespace Cake\ElasticSearch\TestSuite;
+
+use Cake\Datasource\ConnectionManager;
+use Cake\ElasticSearch\Datasource\Connection;
+use Elastica\Mapping as ElasticaMapping;
+use RuntimeException;
+
+/**
+ * Create indexes and mappings for test cases.
+ *
+ * Index definition files should return an array of indexes
+ * to create. Each index in the array should follow the form of
+ *
+ * ```
+ * [
+ *   'name' => 'articles',
+ *   'mapping' => [...],
+ *   'settings' => [...],
+ * ]
+ * ```
+ *
+ * The `mapping` key should be compatible with Elasticsearch's
+ * mapping API and Elastica.
+ *
+ * The `settings` key can contain Elastica compatible index creation
+ * settings.
+ *
+ * @see http://elastica.io/getting-started/storing-and-indexing-documents.html#define-mapping
+ */
+class MappingGenerator
```

Comment:
Not sure on this name either. I was aiming to match the `SchemaGenerator` we have in cake core.